monix-eval/shared/src/test/scala/monix/eval/TaskStartAndForgetSuite.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2014-2018 by The Monix Project Developers.
+ * See the project homepage at: https://monix.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package monix.eval
+
+import cats.laws._
+import cats.laws.discipline._
+
+import scala.util.Success
+
+object TaskStartAndForgetSuite extends BaseTestSuite {
+
+  test("task.startAndForget works") { implicit sc =>
+    check1 { (i: Int) =>
+      var counter = 0
+      val task = Task {
+        counter = i
+        i
+      }
+      task.startAndForget.runAsync
+      sc.tick()
+
+      counter <-> i
```

Comment:
You're using `ScalaCheck` to repeat the test for random `i: Int` values, however that won't do much for us.

Do a simple test and test that:

1. async boundary is needed
2. execution is actually triggered in a background process (its own thread)

```scala
val task = Task.eval { counter += 1; counter }
val task = for {
  _ <- task.delayExecution(1.second).forkAndForget // first
  _ <- task.delayExecution(1.second).forkAndForget // second
} yield ()

val f = task.runAsync
assertEquals(f.value, Some(Success(()))

// Testing that both got executed in parallel
s.tick(1.second)
assertEquals(counter, 2)
```

Well, this doesn't test that an async boundary is forced, since we are doing it anyway with `delayExecution`, but you get the idea.